src/lib.rs

```diff
@@ -0,0 +1,42 @@
+// Copyright (C) 2020 Kisio Digital and/or its affiliates.
+//
+// This program is free software: you can redistribute it and/or modify it
+// under the terms of the GNU Affero General Public License as published by the
+// Free Software Foundation, version 3.
+
+// This program is distributed in the hope that it will be useful, but WITHOUT
+// ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+// FOR A PARTICULAR PURPOSE. See the GNU Affero General Public License for more
+// details.
+
+// You should have received a copy of the GNU Affero General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>
+
+mod attribute;
+pub use attribute::AttributeElementExt;
+mod only_child;
+pub use only_child::OnlyChildElementExt;
+
+use thiserror::Error;
+
+#[derive(Debug, Error)]
+pub enum Error {
+    #[error("Failed to find attribute '{1}' in element '{0}'")]
+    AttributeNotFound(String, String),
+    #[error("No child with expected filter in Element '{0}'")]
```

Comment:
Maybe:
```suggestion
    #[error("No child matching filter in Element '{0}'")]
```
If so, worth changing enum value too, and on Unique also.

Reply:
Done in d37f8887ec9f4822c267e3d11807228ed7ad0eed.